accepted/0026-add-types-metadata-to-package-json.md

```diff
@@ -0,0 +1,102 @@
+# Add Types Metadata to the Registry [Packument](https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md#package-endpoints)
+
+## Summary
+
+Add an unambiguous notation to the derived package.json whether a dependency includes types.
+
+## Motivation
+
+It would be great to know from the NPM API if a library included type definitions. This would allow the npm website, and other tooling to be able to help people make decisions about which dependencies they would like to use and support for types plays into that.
+
+## Detailed Explanation
+
+Working with one of my libraries which ships with both Flow and TypeScript types:
+
+[Package.json](https://github.com/danger/danger-js/blob/e7d9b6a515eabe0488a5f0ec196319c7fbe479b0/package.json)
+
+```json5
+{
+  "name": "danger",
+  "version": "10.1.0",
+  "description": "Unit tests for Team Culture",
+  "main": "distribution/danger.js",
+  "typings": "distribution/danger.d.ts",
+  "bin": {
+    // 
+  }
+  "prettier": {
+    // ...
+  },
+  "scripts": {
+    // ...
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/danger/danger-js.git"
+  },
+  "keywords": [
+    "danger",
+    "ci"
+  ],
+  "author": "Orta Therox",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/danger/danger-js/issues"
+  },
+  "homepage": "https://github.com/danger/danger-js#readme",
+  "devDependencies": {
+    // ...
+  },
+  "dependencies": {
+    // ...
+  },
+}
+```
+
+This `package.json` unambiguously declares that it has TypeScript types. However, a flow user has no idea that I put the time in [to maintain](https://unpkg.com/browse/danger@10.1.0/distribution/danger.js.flow) support for them!
+
+I propose that [the packument version of this package](https://registry.npmjs.org/danger/10.1.0) have a new field `"_types"` which would look like:
+
+```json5
+{
+  // ...
+  "dependencies": {
+    // ...
+  },
+  "gitHead": "66e18daae5286ea64ed49484fd5835685684c156",
+  "_id": "danger@10.0.0",
+  "_types": {
+    "ts": "included",
```

Comment:
why just "included"? It might be nice for a package to be able to point to a DT (or the flow equivalent) package for its types, so users don't have to guess, and if so, that info would presumably be wanted in the packument also?

Reply:
Presumably this could point to the `typings` field if present?

Reply:
right, but what i mean is, it's highly useful to be able to extract "package dep" information from the packument without having to extract the entire tarball; if "typings" is only pointing to relative files, a boolean is sufficient, but if it's pointing to a package, the package name and version would be important too.

Reply:
I'd like it, but it seems unlikely for many to add the additional field to declare that they have a DT/FT representation. I wouldn't be surprised that a lot of projects don't know they have DT/FT files.

That was why I said that I think it probably should be left to another tool to denote whether that types package exists. 

In theory (and just for DefinitelyTyped), because the types are on npm - it might be reasonable for the CLI to ping whether it's equivalent `@types/module` exists in the registry and mark it automatically